atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbConfig.java

```diff
@@ -250,11 +256,35 @@ protected final void check() {
             Preconditions.checkState(areTimeAndLockConfigsAbsent(),
                     "If the timelock block is present, then the lock and timestamp blocks must both be absent.");
         }
+    }
 
+    private void checkLockAndTimestampBlocks() {
         Preconditions.checkState(lock().isPresent() == timestamp().isPresent(),
                 "Lock and timestamp server blocks must either both be present or both be absent.");
     }
 
+    private void checkNamespaces() {
+        if (namespace().isPresent()) {
+            String namespace = namespace().get();
+
+            keyValueService().namespace().ifPresent(kvsNamespace ->
+                    Preconditions.checkState(kvsNamespace.equals(namespace),
+                            "The AtlasDbConfig namespace and the KVS namespace should be the same."));
```

Comment:
This may be confusing, since the KVS namespace config isn't actually set as `namespace`.
We might need to write `keyspace/dbName/Sid`